go/libraries/doltcore/schema/column.go

```diff
@@ -149,8 +149,7 @@ func (c Column) Equals(other Column) bool {
 // kinds of values at the storage layer, but may have different constraints or type parameters.
 func (c Column) Compatible(other Column) bool {
 	return c.Tag == other.Tag &&
-		c.Kind == other.Kind &&
-		c.IsPartOfPK == other.IsPartOfPK
```

Comment:
This is suspect, what's the reasoning for this?

Reply:
The superschema will complain about a tag collision otherwise during. [cc:](https://github.com/dolthub/dolt/blob/83b4aaaf7413f9e49ff8c6c037b12888fe8b95c1/go/libraries/doltcore/schema/super_schema.go#L67)

I also don't think it's correct anymore since primary key changes are possible.

Reply:
We should adjust the super schema logic instead of changing this. changing this here could have unexpected consequences